### PR TITLE
Add h5sample to sample from COMPASOutput files

### PR DESCRIPTION
**Pull request by avivajpeyi**
Add an option to copy only a fraction of the binaries. Eg: 

```
Sampling 100 binaries from detailed_evolution/COMPAS_Output/COMPAS_Output.h5.
Original file summary:


Summary of HDF5 file detailed_evolution/COMPAS_Output/COMPAS_Output.h5
============================================================================================================================================================

File size    : 0.0083 GB
Last modified: 2023-04-05 14:51:08.648755

COMPAS Filename              Columns   Entries   Unique SEEDs
--------------------------   -------   -------   ------------
Run_Details                      438         1
BSE_Common_Envelopes              74         2              2
BSE_Double_Compact_Objects        13         2              2
BSE_RLOF                          39         4              2
BSE_Supernovae                    34         4              2
BSE_System_Parameters            171         2              2


Copying file detailed_evolution/COMPAS_Output/COMPAS_Output.h5
Sampled file summary:


Summary of HDF5 file sampled_compas_out.h5
========================================================================================

File size    : 0.6507 GB
Last modified: 2023-04-06 17:41:00.880481

COMPAS Filename              Columns   Entries   Unique SEEDs
--------------------------   -------   -------   ------------
Run_Details                      438         1
BSE_Common_Envelopes              74       100              2
BSE_Double_Compact_Objects        13       100              2
BSE_RLOF                          39       200              2
BSE_Supernovae                    34       200              2
BSE_System_Parameters            171       100              2

``` 

```

Sampling 1 (50.00%) binaries from detailed_evolution/COMPAS_Output/COMPAS_Output.h5

Summary of HDF5 file sampled_compas_out.h5
========================================================================================

File size    : 0.6507 GB
Last modified: 2023-04-06 17:41:01.604485

COMPAS Filename              Columns   Entries   Unique SEEDs
--------------------------   -------   -------   ------------
Run_Details                      438         1
BSE_Common_Envelopes              74         1              1
BSE_Double_Compact_Objects        13         1              1
BSE_RLOF                          39         2              1
BSE_Supernovae                    34         2              1
BSE_System_Parameters            171         1              1

```

### CODE REVIEW

File: compas_python_utils/h5copy.py [hunk 1]
```diff
@@ -115,7 +117,7 @@
 The commandline option '-r' specifies whether recursion is enabled for directory processing, and if
 it is, to what depth:
 
-    if the -r option is not present, recursion is not enabled and only files in the specified working
```
Comment (avivajpeyi) [inline compas_python_utils/h5copy.py]: urg i think my silly IDE removed this by accident

**Comment by avivajpeyi** [comment]
After chatting with Jeff he pointed out that this option may be better suited in a separate file. I agree!

**Comment by avivajpeyi** [comment]
im trying to use some autodoc tools to auto-gen the documentation for the h5sample module that I've drafted.

building the docs locally, the files are generating: 
Screenshot from 2023-04-06 16-30-13


The online build via read-the-docs is getting messed up cause I cant get RTD to install the Python package before building... 

Screenshot from 2023-04-06 16-45-36


The docs don't seem to be being built from a GitHub workflow but are triggered via RTD itself?

**Comment by avivajpeyi** [comment]
Adding a config `readthedocs.yml` to help specify the location of the python package...

**Comment by avivajpeyi** [comment]
Woohoo! Docs also work 
[See this](https://compas--954.org.readthedocs.build/en/954/pages/User%20guide/Post-processing/post-processing-h5sample.html)

**Comment by ilyamandel** [comment]
Very nice, @avivajpeyi!  But I am leaving this one for @jeffriley to review (when he has time). :)

**Comment by ilyamandel** [comment]
Ping to @jeffriley . :-)

**Review by jeffriley — APPROVED** [review]
Looks good - thanks Avi

**Pull request merged**